src/builtins/fish_indent.rs

```diff
@@ -0,0 +1,1200 @@
+            future_feature_flags::set_from_string(s.as_utfstr());
+        }
+    }
+
+    do_indent(&mut streams, args)
+}
+
+pub fn fish_indent(_parser: &Parser, streams: &mut IoStreams, args: &mut [&wstr]) -> Option<c_int> {
+    let args = args.iter_mut().map(|x| x.to_owned()).collect();
+    Some(do_indent(streams, args))
+}
+
+fn do_indent(streams: &mut IoStreams, args: Vec<WString>) -> i32 {
+    // Types of output we support
+    #[derive(Eq, PartialEq)]
+    enum OutputType {
+        PlainText,
+        File,
+        Ansi,
+        PygmentsCsv,
+        Check,
+        Html,
+    }
+
+    let mut output_type = OutputType::PlainText;
+    let mut output_location = L!("");
+    let mut do_indent = true;
+    let mut only_indent = false;
+    let mut only_unindent = false;
+    // File path for debug output.
+    let mut debug_output = None;
+
+    let short_opts: &wstr = L!("+d:hvwicD:");
+    let long_opts: &[WOption] = &[
+        wopt(L!("debug"), ArgType::RequiredArgument, 'd'),
+        wopt(L!("debug-output"), ArgType::RequiredArgument, 'o'),
+        wopt(L!("debug-stack-frames"), ArgType::RequiredArgument, 'D'),
+        wopt(L!("dump-parse-tree"), ArgType::NoArgument, 'P'),
+        wopt(L!("no-indent"), ArgType::NoArgument, 'i'),
+        wopt(L!("only-indent"), ArgType::NoArgument, '\x04'),
+        wopt(L!("only-unindent"), ArgType::NoArgument, '\x05'),
+        wopt(L!("help"), ArgType::NoArgument, 'h'),
+        wopt(L!("version"), ArgType::NoArgument, 'v'),
+        wopt(L!("write"), ArgType::NoArgument, 'w'),
+        wopt(L!("html"), ArgType::NoArgument, '\x01'),
+        wopt(L!("ansi"), ArgType::NoArgument, '\x02'),
+        wopt(L!("pygments"), ArgType::NoArgument, '\x03'),
+        wopt(L!("check"), ArgType::NoArgument, 'c'),
+    ];
+
+    let mut shim_args: Vec<&wstr> = args.iter().map(|s| s.as_ref()).collect();
+    let mut w = WGetopter::new(short_opts, long_opts, &mut shim_args);
+
+    while let Some(c) = w.next_opt() {
+        match c {
+            'P' => DUMP_PARSE_TREE.store(true),
+            'h' => {
+                print_help("fish_indent");
+                return STATUS_CMD_OK.unwrap();
+            }
+            'v' => {
+                streams.out.appendln(wgettext_fmt!(
+                    "%s, version %s",
+                    PROGRAM_NAME.get().unwrap(),
+                    crate::BUILD_VERSION
+                ));
+                return STATUS_CMD_OK.unwrap();
+            }
+            'w' => output_type = OutputType::File,
+            'i' => do_indent = false,
+            '\x04' => only_indent = true,
+            '\x05' => only_unindent = true,
+            '\x01' => output_type = OutputType::Html,
+            '\x02' => output_type = OutputType::Ansi,
+            '\x03' => output_type = OutputType::PygmentsCsv,
+            'c' => output_type = OutputType::Check,
+            'd' => {
+                activate_flog_categories_by_pattern(w.woptarg.unwrap());
```

Comment:
This currently allows activating debug options globally by running `fish_indent`.

I'm not sure we currently have anything else to do that?

Reply:
I've removed these - we would have to do some work to make them *not* affect the running fish session, and they don't really apply to the fish_indent itself.

If you want to debug how some script parses, you can use `fish --no-execute --debug ...`.